### ISSUE TITLE
Data folder corrupted, how to retrieve passwords from database?

### ISSUE THREAD

**Issue by moretocome** [marius-wieschollek/passwords#114]
Not an issue, but I hope you will help me anyway.

I run a personal Nextcloud 15 with Passwords 2019.1.0 at least or possibly 2019.2.0 on a Raspberry Pi operating of USB drives.

The drive containing my Nextcloud data folder (exclusively) has gone corrupt and is not mountable. I don't have backup of the folder, but I do have the database and the webroot which is on a separate drive.

If I build a fresh Nextcloud and import the ```*prefix*_passwords_*``` tables from the old database to the new one before I activate the passwords app, will I be able to retrieve my passwords? Or is there a better approach?

Help will be greatly appreciated.

**Comment by marius-wieschollek** [comment]
If you have the password tables as well as the `*prefix*_appdata` and `*prefix*_preferences` tables, you should be able to restore your data.

**Comment by moretocome** [comment]
Thanks a lot for your reply!

I have the full database including ```*prefix*_appdata``` and ```*prefix*_preferences```.

But how should I restore them? I am guessing that restoring ```*prefix*_appdata``` in the database of a new installation will cause problems.

**Comment by marius-wieschollek** [comment]
You could also use your old instance to create a backup of the passwords database.
To do so, just point the Nextcloud data directory somewhere else. then take a look at the [backup manual](https://git.mdns.eu/nextcloud/passwords/wikis/Administrators/Backups#creating-backups) to create a new backup and move it to your new instance.

If that is not possible you can just fetch all entries from the tables that have the appid `passwords`. If you move them over to the new instance everything should work.

**Comment by moretocome** [comment]
Thanks for your help @marius-wieschollek :1st_place_medal: 

I installed a fresh nextcloud and imported the old ```*prefix*_passwords_*``` to the new db, before I activated the app.

I then tried to access the passwords, but I just saw a spinning disc, at the log showed me ```Error | passwords | Exception: HMAC does not match.```

Guided by your comments I checked ```*prefix*_appdata``` and ```*prefix*_preferences``` and updated the values of ```SSEv1UserKey``` (one in ```*prefix*_appdata``` and one pr. user in ```*prefix*_preferences```) with the keys from the old db, and now I have access to all of my passwords again! Thank you very much!
